src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -164,12 +181,21 @@ ucc_status_t ucc_tl_ucp_allgather_knomial_start(ucc_coll_task_t *coll_task)
                                    ucc_dt_size(args->dst.info.datatype), rank,
                                    size, radix);
     if (!UCC_IS_INPLACE(*args)) {
-        status = ucc_mc_memcpy(
-            PTR_OFFSET(args->dst.info.buffer, offset), args->src.info.buffer,
-            args->src.info.count * ucc_dt_size(args->src.info.datatype),
-            args->dst.info.mem_type, args->src.info.mem_type);
-        if (UCC_OK != status) {
-            return status;
+        status = ucc_coll_task_get_executor(&task->super, &exec);
```

Comment:
similar memcpy is in the allgather_ring

Reply:
this PR only changes knomial allreduce and sra allreduce, will do changes for other collective later